test/async-emitter.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+
+const { AsyncEmitter } = require('..');
+const metatests = require('metatests');
+
+metatests.test('AsyncEmitter on/emit', async test => {
+  const ae = new AsyncEmitter();
+  let emitted = false;
+
+  ae.on('e1', async (a, b, c, d) => {
+    test.strictSame(a, 1);
+    test.strictSame(b, 2);
+    test.strictSame(c, 3);
+    test.strictSame(d, 4);
+    emitted = true;
+  });
+
+  await ae.emit('e1', 1, 2, 3, 4);
+
+  test.strictSame(emitted, true);
+  test.end();
+});
+
+metatests.test('AsyncEmitter once', async test => {
+  const ae = new AsyncEmitter();
+  let count = 0;
+
+  const f1 = test.mustCall(async () => {
+    count++;
+  });
+
+  const f2 = test.mustCall(async () => {
+    count++;
+  });
+
+  ae.once('e1', f1);
+  ae.once('e1', f2);
+  ae.emit('e1');
+  ae.emit('e1');
+
+  test.strictSame(count, 2);
+
+  test.end();
+});
```

Comment:
To give an example:
```js
metatests.test('AsyncEmitter once', async test => {
  const ae = new AsyncEmitter();

  ae.once('e1', test.mustCall());
  ae.once('e1', test.mustCall());

  await ae.emit('e1');
  await ae.emit('e1');
});
```

`test.mustCall()` by default checks that there was exactly one call. You can check for other number of calls with `test.mustCall(() => {}, 3)`.

Reply:
Thanks, I see `test.mustCall()` requires no first argument in all cases?

Reply:
Thanks to this changes I have found another memory leak https://github.com/metarhia/metasync/pull/427/commits/fae8577248d3106042a49fed6880d6f383d46ac8

Reply:
Yes, the first argument is optional, if not passed `() => {}` will be used.